### PR TITLE
Update plugin BOM and jenkins version

### PR DESCRIPTION
**Pull request by car-roll**
This PR refiles the BOM portion of #498 as it was reverted in #501 due to windows CI failures.

Failures generated are `filename too long`. See: https://github.com/jenkinsci/pipeline-model-definition-plugin/runs/5580211853

See also: https://github.com/jenkinsci/pipeline-model-definition-plugin/pull/499

cc @basil

### CODE REVIEW

**Comment by basil** [comment]
Let's see if just updating the parent POM as in #503 (ignoring the BOM update and version bump) is sufficient to keep the tests stable and for my Java 17 work.

**Comment by car-roll** [comment]
I am currently testing on my local windows env to see if I can reproduce

**Review by basil — CHANGES_REQUESTED** [review]
Build failed. Suggest closing this PR in favor of #503.

**Comment by car-roll** [comment]
Looks like enable `longpaths` worked. Only did it for one test and it passed all other tests (most likely because of `--global`. Will need to clean this up properly. Either call it once for only windows, or just call it always.

**Comment by car-roll** [comment]
going to keep this PR open to see if we can resolve the BOM bump issues

**Review by jglick — APPROVED** [review]

**Review by basil — APPROVED** [review]

**Comment by jglick** [comment]
Also see #476, #504, #487 (override should be removed), and probably others.

**Comment by jglick** [comment]
Hmm, seems like #503 should have caused some of those to be automatically closed? :eyes:

File: pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/LibrariesTest.java [hunk 1]
```diff
@@ -161,6 +161,8 @@ public void librariesDirectiveWithOutsideVarAndFunc() throws Exception {
     @Test
     public void folderLibraryParsing() throws Exception {
         otherRepo.init();
+        // longpaths to support longer filenames for windows. Only needs to be set once.
+        otherRepo.git("config", "--global", "core.longpaths", "true");
```
Comment (jglick) [inline pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/LibrariesTest.java:165]: Can be reverted now that https://github.com/jenkins-infra/pipeline-library/pull/334 is merged?

Comment (car-roll) [reply pipeline-model-definition/src/test/java/org/jenkinsci/plugins/pipeline/modeldefinition/LibrariesTest.java:165]: Filed #506

**Pull request merged**